src/action.ts

```diff
@@ -129,31 +276,25 @@ async function downloadGitHubBinary(
   const pathInCache = toolCache.find(repo, tag);
   if (pathInCache) {
     core.info(`Found ${owner}/${repo} version ${tag} in cache!`);
-
     return { folder: pathInCache, name: binaryName };
   }
 
   core.info(`Downloading ${owner}/${repo} version ${tag}`);
 
-  const getReleaseResponse = await octokit.rest.repos.getReleaseByTag({
-    owner,
-    repo,
-    tag,
-  });
+  const release = await scmProvider.getReleaseByTag(owner, repo, tag);
 
   const re = new RegExp(`${osPlatform()}.*${arch()}`);
-  const asset = getReleaseResponse.data.assets.find((obj: any) => re.test(obj.name));
+  const asset = release.assets.find((obj: any) => re.test(obj.name));
```

Comment:
This is where we need that type on the return of getReleaseByTag -- there's no guarantee github and gitlab both have a name property from their apis